cranelift/codegen/src/opts/vector.isle

```diff
@@ -1,8 +1,77 @@
-;; Lift a splat outside of an int-to-float conversion to try to open up
+;; For various ops lift a splat outside of the op to try to open up
 ;; optimization opportunities with scalars.
 ;;
-;; NB: this is also required for correctness at this time due to #6562
+;; NB: for int-to-float conversion op this simplification is also
+;; required for correctness at this time due to #6562
 (rule (simplify (fcvt_from_uint float_vector_ty (splat _ x)))
       (splat float_vector_ty (fcvt_from_uint (lane_type float_vector_ty) x)))
 (rule (simplify (fcvt_from_sint float_vector_ty (splat _ x)))
       (splat float_vector_ty (fcvt_from_sint (lane_type float_vector_ty) x)))
+
+(rule (simplify (band ty (splat ty x) (splat ty y)))
+      (splat ty (band (lane_type ty) x y)))
+
+(rule (simplify (bor ty (splat ty x) (splat ty y)))
+      (splat ty (bor (lane_type ty) x y)))
+
+(rule (simplify (bxor ty (splat ty x) (splat ty y)))
+      (splat ty (bxor (lane_type ty) x y)))
+
+(rule (simplify (bnot ty (splat ty x)))
+      (splat ty (bnot (lane_type ty) x)))
+
+(rule (simplify (iadd ty (splat ty x) (splat ty y)))
+      (splat ty (iadd (lane_type ty) x y)))
+
+(rule (simplify (isub ty (splat ty x) (splat ty y)))
+      (splat ty (isub (lane_type ty) x y)))
+
+(rule (simplify (imul ty (splat ty x) (splat ty y)))
+      (splat ty (imul (lane_type ty) x y)))
+
+(rule (simplify (smulhi ty (splat ty x) (splat ty y)))
+      (splat ty (smulhi (lane_type ty) x y)))
+
+(rule (simplify (umulhi ty (splat ty x) (splat ty y)))
+      (splat ty (umulhi (lane_type ty) x y)))
+
+(rule (simplify (ineg ty (splat ty x)))
+      (splat ty (ineg (lane_type ty) x)))
+
+(rule (simplify (iabs ty (splat ty x)))
+      (splat ty (iabs (lane_type ty) x)))
+
+(rule (simplify (popcnt ty (splat ty x)))
+      (splat ty (popcnt (lane_type ty) x)))
+
+(rule (simplify (smin ty (splat ty x) (splat ty y)))
+      (splat ty (smin (lane_type ty) x y)))
+
+(rule (simplify (umin ty (splat ty x) (splat ty y)))
+      (splat ty (umin (lane_type ty) x y)))
+
+(rule (simplify (smax ty (splat ty x) (splat ty y)))
+      (splat ty (smax (lane_type ty) x y)))
+
+(rule (simplify (umax ty (splat ty x) (splat ty y)))
+      (splat ty (umax (lane_type ty) x y)))
+
+(rule (simplify (avg_round ty (splat ty x) (splat ty y)))
+      (splat ty (avg_round (lane_type ty) x y)))
```

Comment:
I think `avg_round` might also not have a scalar substitute at least the [docs](https://docs.rs/cranelift-codegen/latest/cranelift_codegen/ir/trait.InstBuilder.html#method.avg_round) don't claim it does.

Reply:
Thanks for spotting that. Removed it.